os_stub/cryptlib_mbedtls/pk/x509.c

```diff
@@ -36,6 +37,52 @@ static const uint8_t m_libspdm_oid_organization_name[] = OID_ORGANIZATION_NAME;
 static const uint8_t m_libspdm_oid_ext_key_usage[] = OID_EXT_KEY_USAGE;
 static const uint8_t m_libspdm_oid_basic_constraints[] = OID_BASIC_CONSTRAINTS;
 
+typedef struct {
+    const char *name; /* String representation of AttributeType, e.g.
+                       * "CN" or "emailAddress". */
+    size_t name_len;  /* Length of 'name', without trailing 0 byte. */
+    const char *oid;  /* String representation of OID of AttributeType,
+                       * as per RFC 5280, Appendix A.1. */
+    size_t oid_len;
+    int default_tag;  /* The default character encoding used for the
+                       * given attribute type, e.g.
+                       * MBEDTLS_ASN1_UTF8_STRING for UTF-8. */
+} libspdm_x509_subject_descriptor_t;
+
+#define ADD_STRLEN( s )     s, sizeof( s ) - 1
+
+/*
+ * EC public keys:
+ *  SubjectPublicKeyInfo  ::=  SEQUENCE  {      1 + 2
+ *    algorithm         AlgorithmIdentifier,    1 + 1 (sequence)
+ *                                            + 1 + 1 + 7 (ec oid)
+ *                                            + 1 + 1 + 9 (namedCurve oid)
+ *    subjectPublicKey  BIT STRING              1 + 2 + 1               [1]
+ *                                            + 1 (point format)        [1]
+ *                                            + 2 * ECP_MAX (coords)    [1]
+ *  }
+ */
+#define ECP_PUB_DER_MAX_BYTES   ( 30 + 2 * MBEDTLS_ECP_MAX_BYTES )
+
+/*
+ * RSA public keys:
+ *  SubjectPublicKeyInfo  ::=  SEQUENCE  {          1 + 3
+ *       algorithm            AlgorithmIdentifier,  1 + 1 (sequence)
+ *                                                + 1 + 1 + 9 (rsa oid)
+ *                                                + 1 + 1 (params null)
+ *       subjectPublicKey     BIT STRING }          1 + 3 + (1 + below)
+ *  RSAPublicKey ::= SEQUENCE {                     1 + 3
+ *      modulus           INTEGER,  -- n            1 + 3 + MPI_MAX + 1
+ *      publicExponent    INTEGER   -- e            1 + 3 + MPI_MAX + 1
+ *  }
+ */
+#define RSA_PUB_DER_MAX_BYTES   ( 38 + 2 * MBEDTLS_MPI_MAX_SIZE )
+
+#define LIBSPDM_MAX_PUBKEY_DER_BUFFER_SIZE (RSA_PUB_DER_MAX_BYTES > ECP_PUB_DER_MAX_BYTES ? \
+                                            RSA_PUB_DER_MAX_BYTES : ECP_PUB_DER_MAX_BYTES )
+
+#define LIBSPDM_MAX_SUBJECT_BUFFER_SIZE 1024
```

Comment:
why 1024?

Reply:
Thanks. I have used the `MBEDTLS_X509_MAX_DN_NAME_SIZE` to replace it.